go-controller/pkg/clustermanager/secondary_network_cluster_manager.go

```diff
@@ -0,0 +1,138 @@
+package clustermanager
+
+import (
+	"github.com/containernetworking/cni/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+
+	ovncnitypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	nad "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/network-attach-def-controller"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+// secondaryNetworkClusterManager object manages the multi net-attach-def controllers.
+// It implements networkAttachDefController.NetworkControllerManager and can be used
+// by NetAttachDefinitionController to add and delete NADs.
+type secondaryNetworkClusterManager struct {
+	// net-attach-def controller handle net-attach-def and create/delete network controllers
+	nadController *nad.NetAttachDefinitionController
+	ovnClient     *util.OVNClusterManagerClientset
+	watchFactory  *factory.WatchFactory
+	stopChan      chan struct{}
+}
+
+func newSecondaryNetworkClusterManager(ovnClient *util.OVNClusterManagerClientset,
+	wf *factory.WatchFactory, recorder record.EventRecorder) *secondaryNetworkClusterManager {
+	klog.Infof("Creating new multi network cluster manager")
+	sncm := &secondaryNetworkClusterManager{
+		ovnClient:    ovnClient,
+		watchFactory: wf,
+		stopChan:     make(chan struct{}),
+	}
+	sncm.nadController = nad.NewNetAttachDefinitionController(
+		"cluster-manager", sncm, ovnClient.NetworkAttchDefClient, recorder)
+	return sncm
+}
+
+// Start starts the secondary layer3 controller, handles all events and creates all needed logical entities
+func (sncm *secondaryNetworkClusterManager) Start() error {
+	klog.Infof("Starting cluster-manager/net-attach-def-controller")
+	go func() {
+		// nadController Run blocks until the stopChan is closed.
+		if err := sncm.nadController.Run(sncm.stopChan); err != nil {
+			klog.Errorf("Nad controll run failed %v", err)
```

Comment:
nit: typo `Nad controll`

I am thinking though, in the interim,  that we could receive the cancel context here call it if Run fails.